NorthstarDLL/keyvalues.cpp

```diff
@@ -568,7 +569,7 @@ KeyValues* KeyValues::GetFirstSubKey() const
 //-----------------------------------------------------------------------------
 KeyValues* KeyValues::GetNextKey() const
 {
-	assert(this, "Member function called on NULL KeyValues");
+	assert(this);

```

Comment:
Should not be throwing away the error messages here. Either make a new macro or do something else.

Reply:
Are you saying to make a new macro to replicate the functionality of the non-standard `assert` or a macro to only use the non-standard `assert` in MSVC and throw away the error message in gcc?

I'm unsure what the non-standard call actually did, because there's no documentation for an assert that takes a second parameter (https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/assert-macro-assert-wassert?view=msvc-170) and the macro appears to be defined as
```
#define assert(expression) (void)(                                                       \
            (!!(expression)) ||                                                              \
            (_wassert(_CRT_WIDE(#expression), _CRT_WIDE(__FILE__), (unsigned)(__LINE__)), 0) \
        )
```
from `C:\Program Files (x86)\Windows Kits\10\Include\10.0.22000.0\ucrt\assert.h`

If you can point to some documentation and clarify what the desired change is I can do that.

Reply:
@emma-miler 

https://en.cppreference.com/w/cpp/error/assert

> There is no standardized interface to add an additional message to assert errors.
> ...
> The implementation of assert in [Microsoft CRT](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/assert-macro-assert-wassert?view=msvc-160) does not conform to C++11 and later revisions, because its underlying function (_wassert) takes neither __func__ nor an equivalent replacement.